app/models/proxy.rb

```diff
@@ -278,7 +280,11 @@ def publish_events
     nil
   end
 
-  DEPLOYMENT_OPTION_CHANGED = ->(record) { record.changed_attributes.key?(:deployment_option) }
+  DEPLOYMENT_OPTION_CHANGED = ->(record) {
+    raise 'crap' if record.saved_changes.key?(:deployment_option) != record.changed_attributes.key?(:deployment_option)
+
+    record.saved_change_to_attribute?(:deployment_option)
+  }
 
   def deployment_option_changed?
     [ self, service ].any?(&DEPLOYMENT_OPTION_CHANGED)
```

Comment:
I think I've tried every new method in every scenario and `saved_change_to_attribute?` is the only that passes all tests and scenarios. I will double check this one just in case.

Reply:
maybe we can inspect the backlogs when it fails to see where it comes from and whether it is legit.